### PR TITLE
Get Player Skin

### PR DESCRIPTION
**Pull request by Leialoha**
Using RetrieveSkin(Player) will give the latest Pocket Edition Skin, if player has a skin it returns their skin instead (Java Players).

### CODE REVIEW

File: spigot/src/main/java/com/github/camotoy/geyserskinmanager/spigot/listener/SpigotPlatformEventListener.java [hunk 1]
```diff
@@ -82,7 +88,45 @@ protected void uploadOrRetrieveSkin(MinecraftProfileWrapper profile, Player play
             }
         }
     }
-
+    
+    public String[] RetrieveSkin(Player player) {
+
+        PlayerProfile playerProfile = player.getPlayerProfile();
```
Comment (Camotoy) [inline spigot/src/main/java/com/github/camotoy/geyserskinmanager/spigot/listener/SpigotPlatformEventListener.java:94]: This cannot be here - it's a Paper-only property, and thus will fail on Spigot. Look at my `MinecraftProfileWrapper` class for an idea of how to work around this.

File: spigot/src/main/java/com/github/camotoy/geyserskinmanager/spigot/listener/SpigotPlatformEventListener.java [hunk 2]
```diff
@@ -82,7 +88,45 @@ protected void uploadOrRetrieveSkin(MinecraftProfileWrapper profile, Player play
             }
         }
     }
-
+    
+    public String[] RetrieveSkin(Player player) {
```
Comment (Camotoy) [inline spigot/src/main/java/com/github/camotoy/geyserskinmanager/spigot/listener/SpigotPlatformEventListener.java:92]: `retrieveSkin` is the correct formatting.

File: spigot/src/main/java/com/github/camotoy/geyserskinmanager/spigot/listener/SpigotPlatformEventListener.java [hunk 3]
```diff
@@ -82,7 +88,45 @@ protected void uploadOrRetrieveSkin(MinecraftProfileWrapper profile, Player play
...
+            return new String[]{setSkin.getJavaSkinValue(), setSkin.getJavaSkinSignature()};
+        }
+        
+        /*
+            Default Steve Skin, Value and Signature
+            This is Provided if setSkin is Null
+            (Just to bypass any errors)
+        */
+        
+        String value = "ewogICJ0aW1lc3RhbXAiIDogMTYxMjY0NzA0MzEwOCwKICAicHJvZmlsZUlkIiA6ICJmYmYyMmIyMjNjN2E0NzA0OGYwM2U0MzVhZGFhNGVhZiIsCiAgInByb2ZpbGVOYW1lIiA6ICJCZWFyZGVkU3RldmUiLAogICJzaWduYXR1cmVSZXF1aXJlZCIgOiB0cnVlLAogICJ0ZXh0dXJlcyIgOiB7CiAgICAiU0tJTiIgOiB7CiAgICAgICJ1cmwiIDogImh0dHA6Ly90ZXh0dXJlcy5taW5lY3JhZnQubmV0L3RleHR1cmUvMWE0YWY3MTg0NTVkNGFhYjUyOGU3YTYxZjg2ZmEyNWU2YTM2OWQxNzY4ZGNiMTNmN2RmMzE5YTcxM2ViODEwYiIKICAgIH0KICB9Cn0=";
```
Comment (Camotoy) [inline spigot/src/main/java/com/github/camotoy/geyserskinmanager/spigot/listener/SpigotPlatformEventListener.java:124]: I think a cleaner implementation would just be to return null and let the plugin deal with it.

**Review by Camotoy — CHANGES_REQUESTED** [review]

**Comment by Leialoha** [comment]
Updated to only get skin data that was saved, otherwise returned with null.

**Comment by Camotoy** [comment]
At this point I'm going to close this since I don't think GeyserSkinManager should be used as API; use Floodgate at this point. Thank you anyway!

**Pull request closed without merging**